Bug #105054.

We do not want the build root in the linked perl module's RUNPATH, so strip
paths containing PORTAGE_TMPDIR if its set.

--- perl-5.8.7/lib/ExtUtils/MM_Unix.pm	2005-09-10 14:06:59.000000000 +0200
+++ perl-5.8.7.az/lib/ExtUtils/MM_Unix.pm	2005-09-10 15:25:52.000000000 +0200
@@ -1915,6 +1915,13 @@
 	    # LD_RUN_PATH now computed by ExtUtils::Liblist
 	    ($self->{EXTRALIBS},  $self->{BSLOADLIBS}, 
              $self->{LDLOADLIBS}, $self->{LD_RUN_PATH}) = @libs;
+	    # We do not want the build root in RPATH
+	    if (exists $ENV{PORTAGE_TMPDIR}) {
+	      # If we have PORTAGE_TMPDIR set, strip that, as just testing for
+	      # /usr and /opt might not be sufficient
+	      $self->{LD_RUN_PATH} = join ':', grep !/^\Q$ENV{PORTAGE_TMPDIR}/,
+	                             split /:/, $self->{LD_RUN_PATH};
+	    }
 	    last;
 	}
     }
